pkg/blob/controllerserver.go

```diff
@@ -779,6 +788,53 @@ func (d *Driver) copyVolume(ctx context.Context, req *csi.CreateVolumeRequest, a
 	}
 }
 
+func (d *Driver) authorizeAzcopyBySecurityPrincipal() error {
+	if len(d.cloud.Config.AADClientSecret) > 0 {
+		klog.V(2).Infof("use service principal to authorize azcopy")
+		os.Setenv("AZCOPY_AUTO_LOGIN_TYPE", "SPN")
+		if err := os.Setenv("AZCOPY_AUTO_LOGIN_TYPE", "SPN"); err != nil {
+			klog.Errorf("failed to set AZCOPY_AUTO_LOGIN_TYPE=SPN, error: %v", err)
+			return err
+		}
+		if d.cloud.Config.AADClientID == "" || d.cloud.Config.AADClientSecret == "" || d.cloud.Config.TenantID == "" {
```

Comment:
`d.cloud.Config.TenantID` is optional here

Reply:
only set it when it is not empty